azure_img_utils/cli/cli_utils.py

```diff
@@ -0,0 +1,319 @@
+# -*- coding: utf-8 -*-
+
+"""Azure image utils cli module utils."""
+
+# Copyright (c) 2022 SUSE LLC
+#
+# This file is part of azure_img_utils. azure_img_utils provides an
+# api and command line utilities for handling images in the Azure Cloud.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import click
+import logging
+import os
+import sys
+import yaml
+
+from collections import namedtuple, ChainMap
+
+
+default_config_dir = os.path.expanduser('~/.config/azure_img_utils/')
+default_profile = 'default'
+
+config_defaults = {
+    'config_dir': default_config_dir,
+    'profile': default_profile,
+    'log_level': logging.INFO,
+    'no_color': False,
+    'sas_token': None,
```

Comment:
ok, you think it might be better to allow only credentials-file option as an argument in a CLI for now and simplify the arg management logics if the options are not likely to be used, right? Seems reasonable to me, I'll simplify it and we can add it later if it's a requirement...

Reply:
Yep, that sounds good. The dictionary credentials option in the class is mainly there for ease of use when using the class as SDK. Thus it probably doesn't need exposure to CLI.